internal/app/backend/backend_service.go

```diff
@@ -72,52 +73,101 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 		return status.Error(codes.InvalidArgument, ".profile is required")
 	}
 
-	resultChan := make(chan mmfResult, len(req.GetProfiles()))
-
-	var syncID string
-	var err error
-
-	startTime := time.Now()
-
-	syncID, err = s.synchronizer.register(stream.Context())
+	syncStream, err := s.synchronizer.synchronize(stream.Context())
 	if err != nil {
 		return err
 	}
 
-	telemetry.RecordNUnitMeasurement(stream.Context(), mRegisterPhase, time.Since(startTime).Milliseconds())
-	startTime = time.Now()
+	errors := make(chan error, 2)
+	mmfCtx, cancelMmfs := context.WithCancel(stream.Context())
+	startMmfs := func() {
+		resultChan := make(chan mmfResult, len(req.GetProfiles()))
 
-	err = doFetchMatchesReceiveMmfResult(stream.Context(), s.mmfClients, req, resultChan)
-	if err != nil {
-		return err
-	}
+		err := doFetchMatchesReceiveMmfResult(mmfCtx, s.mmfClients, req, resultChan)
+		if err != nil {
+			// TODO: Log but continue case where mmfs were canceled once fully
```

Comment:
This is a reminder to myself that once this is fully streaming (as in, not blocking on the MMFs before sending any matches to the syncrhonizer), that I need to log in the case where the context was canceled, but continue with the overall call instead of just erroring out.